File: LeetCode/May challenge/Day 26/Day_26.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        count = 0
        mid_res = 0
        max_len = 0
        temp = {0 : -1}
        for i, num in enumerate(nums):
            if num == 0:
                count += 1
            else:
                count -= 1
            if count in temp:
                mid_res = i - temp[count]
            else:
                temp[count] = i
            max_len = max(mid_res, max_len)
        return max_len

File: LeetCode/May challenge/Day 26/Day_26.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        unordered_map<int, int> m;
        m[0]=-1;
        int ans=0, curr=0;
        for(int i=0;i<nums.size();i++)
        {
            curr=curr+ (nums[i]==0? -1:1);
            if(m.count(curr))
                ans= max(ans, i-m[curr]);
            else
                m[curr]=i;
        }
        return ans;
    }
};
